api-registration/server/index.js

```diff
@@ -0,0 +1,63 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2
+    .hash(req.body.password)
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning *`;
+      const params = [req.body.username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const user = result.rows;
+          res.status(201).json(user);
```

Comment:
Sorry, I missed this the first time. The client should receive an object and not an array. Also, the client should not receive the hashed password in the response.

Reply:
I'm a little confused as the client isn't receiving the hashed password and it's also an object and not an array. It's in the first screenshot as well.